src/Core/Validation/FieldValidation/IntFieldValidator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\FieldValidation\NumericFieldValidator;
+
+class IntFieldValidator extends NumericFieldValidator
+{
+    /**
+     * The minimum value for a signed 32-bit integer.
+     * Defined as string instead of int because be cast to a float
+     * on 32-bit systems if defined as an int
+     */
+    private const MIN_32_BIT_INT = '-2147483648';
+
+    /**
+     * The maximum value for a signed 32-bit integer.
+     */
+    private const MAX_32_BIT_INT = '2147483647';
+
+    public function __construct(
+        string $name,
+        mixed $value,
+        ?int $minValue = null,
+        ?int $maxValue = null
+    ) {
+        if (is_null($minValue)) {
+            $minValue = (int) IntFieldValidator::MIN_32_BIT_INT;
+        }
+        if (is_null($maxValue)) {
+            $maxValue = (int) IntFieldValidator::MAX_32_BIT_INT;
+        }
```

Comment:
I'd rather have them "long form", include the _32_ and _64_ in the constant which makes its a little clearer what they represent, and also because the _64_ versions are string, so it would be be confusing having the const type change between classes

Reply:
> I'd rather have them "long form", include the 32 and 64 in the constant which makes its a little clearer what they represent

The PHPDoc can make that clear, I don't think the const name needs to have that level of detail, especially given it a) results in duplicated code and b) means you can get the `MIN_32_BIT_INT` on a subclass of `IntFieldValidator` which doesn't itself use that constant. Makes more sense for me that `MIN_INT` gives you the min int value for _that class_.

> and also because the 64 versions are string

So are the 32 bit int constants.

Reply:
Updated